manifests/pipecd/templates/deployment.yaml

```diff
@@ -84,6 +84,7 @@ spec:
         {{- include "pipecd.selectorLabels" . | nindent 8 }}
         app.kubernetes.io/component: server
     spec:
+      serviceAccountName: {{ include "pipecd.serviceAccountName" . }}
```

Comment:
@mugioka Since the default value has been changed to `false`, I think we need to add a check condition here.

Reply:
OK!
I added the default value to service account name.
https://github.com/pipe-cd/pipecd/pull/3535/commits/d763a3d3f0a16eccd8adb073d5588164248fc6a6

Reply:
@mugioka I got your point. I think that way is good as well.
But to be honest, I want to introduce the changes to end-users as little as possible.
For existing users, their deployment resource is not containing the `serviceAccountName` but after getting this release, their deployment resource will be updated to add `serviceAccountName: default`.
In most cases, there will be no difference between them but to be safe we should not introduce that change.

So I still prefer adding a check condition over specifying the "default" name as https://github.com/pipe-cd/pipecd/commit/d763a3d3f0a16eccd8adb073d5588164248fc6a6.

WDYT?

Reply:
> In most cases, there will be no difference between them but to be safe we should not introduce that change.

I see, that makes sense.
I will add check statement.

Reply:
Thank you.

Reply:
I added the if statement.

Reply:
Oh, sorry, an unnecessary file was committed.
I will fix it.

Reply:
I fixed it.